### ISSUE TITLE
Lifetime-extended variable incorrectly reported as 'static

### ISSUE THREAD

**Issue by DaniPopes** [rust-lang/rust-analyzer#20510]
**rust-analyzer version**: 0.3.2577-standalone

**rustc version**: rustc 1.91.0-nightly (05f5a58e8 2025-08-19)

**editor or extension**: (eg. VSCode, Vim, Emacs, etc. For VSCode users, specify your extension version; for users of other editors, provide the distribution if applicable)

**relevant settings**: N/A

**repository link (if public, optional)**: N/A

**code snippet to reproduce**:
```rust
fn main() {
    let x = 0usize;
    let _s = match Some(()) {
        Some(()) => &format!("{x}"),
        None => "",
    };
}
```

`_s`' type hint is `&'static str`

<img width="405" height="169" alt="Image" src="https://github.com/user-attachments/assets/cf175672-940f-45cf-b88e-0c2fb6185dcf" />

**Comment by ChayimFriedman2** [comment]
That's because we don't implement lifetimes correctly (or rather: we don't implement lifetimes, period).

To be a bit more technical: we create an inference var for the lifetime and then when it's equated it's inferred to be `'static`, while we should not be using a var here.

**Comment by flodiebold** [comment]
Isn't the problem rather that we don't do a least upper bound and (I assume) instead just equate the types (and hence lifetimes)?

**Comment by ChayimFriedman2** [comment]
This if "fixed", probably by https://github.com/rust-lang/rust-analyzer/pull/20664, but it's probably just irrelevant because that PR made us not handle lifetimes.
